internal/providers/gitlab/manager/webhook_handlers_releases.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+	gitlablib "github.com/xanzy/go-gitlab"
+
+	entmsg "github.com/stacklok/minder/internal/entities/handlers/message"
+	"github.com/stacklok/minder/internal/entities/properties"
+	"github.com/stacklok/minder/internal/events"
+	"github.com/stacklok/minder/internal/providers/gitlab"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+func (m *providerClassManager) handleRelease(l zerolog.Logger, r *http.Request) error {
+	l.Debug().Msg("handling release event")
+
+	releaseEvent := gitlablib.ReleaseEvent{}
+	if err := decodeJSONSafe(r.Body, &releaseEvent); err != nil {
+		return fmt.Errorf("error decoding release event: %w", err)
+	}
+
+	releaseID := releaseEvent.ID
+	if releaseID == 0 {
+		return fmt.Errorf("release event missing ID")
+	}
+
+	tag := releaseEvent.Tag
+	if tag == "" {
+		return fmt.Errorf("merge request event missing IID")
+	}
+
+	rawProjectID := releaseEvent.Project.ID
+	if rawProjectID == 0 {
+		return fmt.Errorf("merge request event missing project ID")
+	}
+
+	// TODO: Should we explicitly handle upcoming/historical releases?
+
+	switch {
+	case releaseEvent.Action == "create":
```

Comment:
These should be constants in the gitlab module but they are missing them. I'll add them.

Reply:
Follow up https://github.com/xanzy/go-gitlab/pull/2032